### ISSUE TITLE
[FEATURE] AI-BI Genie Terraform Support

### ISSUE THREAD

**Issue by pcsrijith** [databricks/terraform-provider-databricks#4082]
Hi Community,

Can we have Terraform support for AI-BI Genie features for Databricks.

### Use-cases
1. Enabling Genie workspace feature at account/workspace levels.
2. Creation of Genie Workspace
3. Tagging of tables in genie workspaces
4. Giving ACLs to users/groups without removing existing roles in genie workspace.
5. Creation of sql queries inside genie workspace.
6. Creation of sql functions inside genie workspace.
7. Add/remove benchmarks in genie workspace.
8. Attaching SQL Warehouse to any new/existing genie workspace.

and any other features which I might have missed logging here.

**Comment by alexott** [comment]
there is no public API for that (yet)

**Comment by alexott** [comment]
Connect with your account team - they can talk with the product team. Terraform team doesn't have insights into the timelines

**Comment by VOVELEE** [comment]
hey @alexott ,

I assume some of the requirements above can be met via https://docs.databricks.com/api/azure/workspace/genie.
Is my understanding correct?

P.S. I also require this in order to properly govern genie spaces on top of workspaces :)

**Comment by alexott** [comment]
There is still no Create/Update API for that.

**Comment by VOVELEE** [comment]
yes, indeed :)
@alexott but what about the following use case - list all genie spaces (list is possible) and assigned mandatory permissions (potentially using databricks_permissions)

**Comment by alexott** [comment]
New resources will be autogenerated as soon as the API is complete, so it's better to wait. Also, I don't see permissions API for genie...

**Comment by VOVELEE** [comment]
:( 
Indeed it looks like inmature implementation in its current state.
Thanks for checking

**Comment by shadzic** [comment]
I am interested in this feature as well.
Doesn't this Set Object Permission in databricks REST API include genie?
https://docs.databricks.com/api/azure/workspace/permissions/set

**Comment by pcsrijith** [comment]
@alexott : I see that create genie space is in Beta phase. Do you think, this can be picked up for curation? [Reference API Link](https://docs.databricks.com/api/azure/workspace/genie/createspace)

**Comment by alexott** [comment]
New resources are automatically generated when the corresponding engineering team decides that the API is stable.

**Comment by nijanthanvijayakumar** [comment]
Hi @alexott,

I recently created a PR for Genie Space support for the Databricks CLI via DABs (https://github.com/databricks/cli/pull/4191). The implementation uses DABs `direct mode` as I realised while working on it that the TF support doesn't exist yet. Local acceptance tests and manual testing against Databricks Free Edition are passing for that PR.

Given that:
- Go SDK already has the Genie API types (CreateSpace, GetSpace, UpdateSpace, TrashSpace)
- TF provider already has the corresponding struct definitions in internal/service/dashboards_tf/

Would a PR adding `databricks_genie_space` resource be accepted?

Understanding the API is still beta, but the building blocks are already in place and there's user demand (this issue + the CLI feature request that led to my PR).

**Comment by nijanthan-dev** [comment]
Hi @alexott,

I recently implemented Genie Space support for the Databricks CLI via DABs (https://github.com/databricks/cli/pull/4191). The implementation uses direct mode because I realised when trying to implement DAB feature that TF support doesn't exist yet for Genie Space. Local acceptance tests and manual testing against Databricks Free Edition are passing from my DABs implementation for direct mode.

Given that:
- Go SDK already has the Genie API types (CreateSpace, GetSpace, UpdateSpace, TrashSpace)
- TF provider already has the corresponding struct definitions in internal/service/dashboards_tf/

Would a PR adding `databricks_genie_space` resource be accepted? I can write unit tests locally but would need CI for integration tests (only have Free Edition access).

Understanding the API is still beta, but the building blocks are in place and there's user demand (this issue + the CLI feature request).

**Comment by alexott** [comment]
No. All new resources are automatically generated from the API spec, when the corresponding team decides it. Raise this issue to your Databricks representative.

**Comment by MigQ2** [comment]
Hi @alexott, I see there is now a [Create Genie Space in the API](https://docs.databricks.com/api/workspace/genie/createspace), also the [ai-dev-kit](https://github.com/databricks/terraform-provider-databricks/issues/4082) already supports creating and updating a Genie space programatically, so I guess it should now be possible to manage a Genie Space using Terraform

Could this issue be tackled now?

**Comment by alexott** [comment]
@MigQ2 all new resources are now automatically generated when the corresponding product team make a decision to support Terraform - please reach your Databricks account team to raise the awareness.

**Comment by pcsrijith** [comment]
@andrewnester Could you please help on this. This issue has been open since Genie was made available. We are unable to productionise our solutions without this. Also, @alexott , requesting you to add the required tags as current one says `needs-api`, which is already GA.

**Comment by pcsrijith** [comment]
Since [CLI v1.3.0](https://github.com/databricks/cli/releases/tag/v1.3.0) is out with Genie support, can we look into this implementation please.
